src/sqlancer/Main.java

```diff
@@ -254,30 +254,36 @@ private void printState(FileWriter writer, StateToReproduce state) {
             this.globalState = globalState;
         }
 
-        public boolean execute(Query q) throws SQLException {
-            globalState.getState().logStatement(q);
-            boolean success = q.execute(globalState);
-            Main.nrSuccessfulActions.addAndGet(1);
-            return success;
-        }
-
-        public ResultSet executeAndGet(Query q) throws SQLException {
-            globalState.getState().logStatement(q);
-            ResultSet result = q.executeAndGet(globalState);
-            Main.nrSuccessfulActions.addAndGet(1);
-            return result;
-        }
-
-        public boolean fillAndExecute(Query q, String template, List<String> fills) throws SQLException {
+        public boolean execute(Query q, String... templateAndFills) throws SQLException {
             globalState.getState().logStatement(q);
-            boolean success = q.fillAndExecute(globalState, template, fills);
+            boolean success;
+            if (templateAndFills.length < 2) {
+                success = q.execute(globalState);
+            } else {
+                String template = templateAndFills[0];
+                List<String> fills = new ArrayList<>();
+                for (int i = 1; i < templateAndFills.length; i++) {
+                    fills.add(templateAndFills[i]);
+                }
+                success = q.fillAndExecute(globalState, template, fills);
+            }
             Main.nrSuccessfulActions.addAndGet(1);
             return success;
         }
 
-        public ResultSet fillAndExecuteAndGet(Query q, String template, List<String> fills) throws SQLException {
+        public ResultSet executeAndGet(Query q, String... templateAndFills) throws SQLException {
```

Comment:
Is this what you had in mind? I could not remove the fill-in-specific methods completely since they use different methods to prepare and execute SQL statements. Instead, I turned the methods in the GlobalState and Main classes into a generic format, and altered the method in the Main class to include a logic to decide which type of method from the Query class to call.

Reply:
It's almost what I had in mind: I would still keep the `String template` parameter, since this is a necessary argument, and have the varargs as an additional parameter. Thanks for incorporating the feedback!

Reply:
Of course! String template is not a necessary argument for the usual query execution, and introducing that could require going into all DBMS implementations and changing that. Currently, `globalStatement.executeStatement(query)` is a valid call, and does not require the query string additionally. Would you like to change the code in each DBMS, then?

Reply:
I looked at the code in greater detail, and I'm not sure if I fully understand. Is the query string in `Query q` left empty when the fills are used? If yes, I think that's undesirable. In this case, I think it would make sense to use the query string to store the template.

Reply:
Just to clarify, then we wouldn't be passing a separate templateString at all which you asked beforehand. We would be using the String as part of the query.

Reply:
Sorry, I think I initially misunderstood how the method should be used.